core/services/offchainreporting2/validate.go

```diff
@@ -0,0 +1,156 @@
+package offchainreporting2
+
+import (
+	"time"
+
+	"github.com/lib/pq"
+
+	"github.com/multiformats/go-multiaddr"
+	"github.com/pelletier/go-toml"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/chains/evm"
+	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+	offchainreporting "github.com/smartcontractkit/libocr/offchainreporting2"
+	"go.uber.org/multierr"
+)
+
+type ValidationConfig interface {
+	Dev() bool
+	OCRBlockchainTimeout() time.Duration
+	OCRContractConfirmations() uint16
+	OCRContractPollInterval() time.Duration
+	OCRContractTransmitterTransmitTimeout() time.Duration
+	OCRDatabaseTimeout() time.Duration
+}
+
+// ValidatedOracleSpecToml validates an oracle spec that came from TOML
+func ValidatedOracleSpecToml(chainSet evm.ChainSet, tomlString string) (job.Job, error) {
+	var jb = job.Job{}
+	var spec job.OffchainReporting2OracleSpec
+	tree, err := toml.Load(tomlString)
+	if err != nil {
+		return jb, errors.Wrap(err, "toml error on load")
+	}
+	// Note this validates all the fields which implement an UnmarshalText
+	// i.e. TransmitterAddress, PeerID...
+	err = tree.Unmarshal(&spec)
+	if err != nil {
+		return jb, errors.Wrap(err, "toml unmarshal error on spec")
+	}
+	err = tree.Unmarshal(&jb)
+	if err != nil {
+		return jb, errors.Wrap(err, "toml unmarshal error on job")
+	}
+	jb.Offchainreporting2OracleSpec = &spec
+	if jb.Offchainreporting2OracleSpec.P2PBootstrapPeers == nil {
+		// Empty but non-null, field is non-nullable.
+		jb.Offchainreporting2OracleSpec.P2PBootstrapPeers = pq.StringArray{}
+	}
+
+	// TODO(#175801038): upstream support for time.Duration defaults in go-toml
```

Comment:
Let's link a card here in the comment pls

Reply:
I think these are stale - we'd move to envconfig if anything to improve this. Just removed them